<launch>
    <!-- 启动 tf_sim -->
    <!-- <node name="tf_sim" pkg="seg_ros" type="tf_sim" output="screen"/> -->

    <!-- 相机z轴的最大视野范围 -->
    <param name="view_field" type="double" value="1.4" />
    <!-- 输入相机的降采样点云数上限 -->
    <param name="imput_num" type="int" value="10000" />
    <!-- 体素滤波的单元格大小 -->
    <param name="grid_size" type="double" value="0.02" />
    <!-- 一级减速距离 -->
    <param name="slow_dis1" type="double" value="1.60" />
    <!-- 二级减速距离 -->
    <param name="slow_dis2" type="double" value="0.80" />
    <!-- 一级减速危险点云个数 -->
    <param name="danger_num1" type="int" value="1000" />
    <!-- 二级减速危险点云个数 -->
    <param name="danger_num2" type="int" value="200" />
    <!-- 机械臂连杆圆柱包络增长比例因子 -->
    <param name="long_factor" type="double" value="0.3" />
    <!-- 机械臂连杆1圆柱包络半径 -->
    <param name="link1_radius" type="double" value="0.065" />
    <!-- 机械臂连杆2圆柱包络半径 -->
    <param name="link2_radius" type="double" value="0.1" />
    <!-- 机械臂连杆3圆柱包络半径 -->
    <param name="link3_radius" type="double" value="0.1" />
    <!-- 机械臂连杆4圆柱包络半径 -->
    <param name="link4_radius" type="double" value="0.1" />
    <!-- 机械臂连杆5圆柱包络半径 -->
    <param name="link5_radius" type="double" value="0.08" />
    <!-- 机械臂连杆6圆柱包络半径 -->
    <param name="link6_radius" type="double" value="0.08" />

    <node name="pc_cam2" pkg="depth2octomap" type="pc_cam2" output = "screen"/>

    <node name="reset_node" pkg="depth2octomap" type="reset_node" output = "screen"/>

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="0.05" />
        <!-- resolution in meters per pixel -->
        <param name="frame_id" type="string" value="base_link" />
        <!-- <param name="frame_id" type="string" value="camera_link" /> -->
        <!-- name of the fixed frame, needs to be "/map" for SLAM -->
        <param name="sensor_model/max_range" value="100.0" />
        <!-- max range / depth resolution of the kinect in meter -->
        <param name="latch" value="true" />
        <!-- max/min height for occupancy map, should be in meters -->
        <!-- <param name="pointcloud_max_z" value="2.0" /> -->
        <!-- <param name="pointcloud_min_z" value="-0.5" /> -->
        <!-- topic from where pointcloud2 messages are subscribed -->
        <!-- <remap from="/cloud_in" to="/pointcloud/output" /> -->
        <remap from="/cloud_in" to="/pointcloud/merged" />
    </node>

    <!-- 启动 pose_pub.py -->
    <node name="pose_pub" pkg="seg_ros" type="pose_pub.py" output="screen" />

    <!-- 启动 on_line -->
    <node name="on_line" pkg="seg_ros" type="on_line" output="screen"/>

</launch>